gateway/client-switchover/README.md

```diff
@@ -0,0 +1,422 @@
+```
+kubectl apply -f gateway-blue.yaml -n confluent
+```
+- Wait for the gateway pods to become READY
+```
+kubectl wait --for=condition=Ready pod -l app=confluent-gateway-blue --timeout=600s -n confluent
+```
+
+#### 2. Deploy Green Gateway [Initially STANDBY]
+- Modify the `streamingDomains` section in the [gateway-green.yaml](./gateway-green.yaml)  to point to your Kafka cluster SASL/PLAIN listener.
+```
+kubectl apply -f gateway-green.yaml -n confluent
+```
+- Wait for the gateway pods to become READY
+```
+kubectl wait --for=condition=Ready pod -l app=confluent-gateway-green --timeout=600s -n confluent
+```
+#### 3. Deploy Loadbalancer Service [Initially pointing to Blue deployment]
+- Modify the port mappings in `spec.ports` corresponding to your source and destination Kafka node id ranges.
+```
+kubectl apply -f loadbalancer-service.yaml -n confluent
+```
+
+- Map the following DNS name to the created loadbalancer: `gateway.example.com`.
+```
+NOTE:
+For the rest of this document, we will be using `gateway.example.com` as the loadbalancer domain name.
+Please replace this appropriately if the configured domain name is different or if you would prefer to use the loadbalancer IP address instead.
+Also, please change the Kafka bootstrap listener port corresponding to this endpoint appropriately, based on the applied loadbalancer service yaml.
+```
+
+### Step 3: Create source and destination Kafka cluster configuration files.
+
+1. Create source cluster configuration: `source-cluster.config`.
+- Modify the `sasl.jaas.config` section with appropriate credentials and the `bootstrap.servers` section with the appropriate endpoint.
+```
+bootstrap.servers=kafka-source:9093
+security.protocol=SASL_PLAINTEXT
+sasl.mechanism=PLAIN
+sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="admin" password="admin-secret";
+```
+
+2. Create destination cluster configuration: `destination-cluster.config`.
+- Modify the `sasl.jaas.config` section with appropriate credentials.
+```
+security.protocol=SASL_PLAINTEXT
+sasl.mechanism=PLAIN
+sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="admin" password="admin-secret";
+```
+
+### Step 4: Test Initial Gateway setup [Loadbalancer pointing to Blue deployment]
+
+1. Test producing messages:
+```
+kafka-console-producer \
+  --bootstrap-server gateway.example.com:9595 \
+  --producer.config source-cluster.config \
+  --topic gateway-blue-test
+```
+2. Test consuming messages:
+```
+kafka-console-consumer \
+  --bootstrap-server gateway.example.com:9595 \
+  --consumer.config source-cluster.config \
+  --topic gateway-blue-test \
+  --from-beginning
+```
+
+### Step 5: Configure Cluster Linking from source to destination Kafka cluster.
+#### NOTE: Please modify the `bootstrap-server` config appropriately for all commands in this section.
+
+#### 1. Create the Cluster Link on the destination cluster
```

Comment:
Since source.config and destination.config are required for CL setup, we can have 3 files

1. source.config (src-admin, src-admin-secret)
2. destination.config (dest-admin, dest-admin-secret)
3. client.properties (usr1, usr1-secret)